compiler/il/OMRNode.hpp

```diff
@@ -546,6 +546,25 @@ class OMR_EXTENSIBLE Node
     */
    bool                   isEAEscapeHelperCall();
 
+   /**
+    * \brief
+    *    Creates a node for storing into an address field of an object
+    *
+    * \parm obj
+    *    Node containing object reference
+    *
+    * \parm symRef
+    *    Symbol reference of the address field
+    *
+    * \parm value
+    *    Value to store
+    *
+    * \return
+    *    Created node
+    *
+    */
+   static TR::Node *      storeToAddressField(TR::Node *obj, TR::SymbolReference *symRef, TR::Node *value);
+
```

Comment:
I only meant it for a field. I think expanding/testing it with arrays will require another PR.

Reply:
Array accesses have a different shape in the trees, they require additional information to create (the index or offset), and they have additional preconditions at runtime: at least a bounds requirement, an alignment requirement if based on offset, and potentially also a runtime type requirement for the array. I think it would make most sense for stores to array elements to be handled by a separate method. In general callers would need to know whether they're storing to a field vs. array element anyway. IMO such a separate method should be implemented as needed in the future, i.e. not as a follow-up just for completeness